lib/common/subredditPost.dart

```diff
@@ -71,24 +78,57 @@ class SubredditPost extends StatelessWidget {
                         ],
                       ),
                     ),
-                    submissionData.isSelf || isViewingPost
+                    widget.submissionData.isSelf || widget.isViewingPost
                         ? Container()
-                        : _buildPostThumbnail(submissionData.preview),
+                        : _buildPostThumbnail(widget.submissionData.preview),
                     Row(
                       mainAxisAlignment: MainAxisAlignment.start,
                       children: [
-                        if (!isViewingPost)
+                        if (!widget.isViewingPost)
                           IconButton(
                             icon: Icon(
                               EvaIcons.messageSquareOutline,
                             ),
                             onPressed: () {
-                              if (onCommentTap != null) onCommentTap();
+                              if (widget.onCommentTap != null)
+                                widget.onCommentTap();
+                            },
+                          ),
+                        if (widget.isLoggedIn != null &&
```

Comment:
I would be very scared if bools end up as null on this project, can you please remove any null checking in regards to `boolean` values?

Reply:
You're right, yesterday I added this check because I had a strange error in the app. Today I have changed a bit the if condition and I don't have the error, so I have removed "widget.isLoggedIn != null".